ci/vm/provision2.sh

```diff
@@ -13,14 +13,14 @@ xpmem_build() {
   git checkout FETCH_HEAD
   ./autogen.sh
   if [[ $OS == *"ubuntu"* ]]; then
-    ./configure --enable-gtest --with-kerneldir=/usr/src/linux-headers-"$(uname -r)"
-    make -s
+    CFLAGS='-Werror -Wall' ./configure --enable-gtest --with-kerneldir=/usr/src/linux-headers-"$(uname -r)"
+    make -s KFLAGS='-Werror -Wno-error=frame-larger-than='
     make check
   elif [[ $OS == *"centos"* ]]; then
     # Build with GCC-8
     scl enable devtoolset-8 -- bash -c "
-      ./configure --enable-gtest --with-kerneldir=/usr/src/kernels/'$(uname -r)'
-      make -s
+      CFLAGS='-Werror -Wall' ./configure --enable-gtest --with-kerneldir=/usr/src/kernels/'$(uname -r)'
+      make -s KFLAGS='-Werror -Wno-error=frame-larger-than='
```

Comment:
can we define the extra CFLAGS in a common place?

Reply:
The variable `CFLAGS` is used for userspace (libxpmem.so and some C tests) and kernel configure detections. KFLAGS is meant for added kernel build flags, so I think we should keep the added options separated. Adding `-Wall` to gtest/userspace is fine, but breaks module build as some warnings are being output.

To simplify, maybe we can add `./configure ... --enable-errors` to activate strict build and add corresponding `-W..` to `CFLAGS` and `KCPPFLAGS`.

Reply:
Stack is consumed by `cpumask_t` (>1kB) and array of 128 `struct page *` (~1kB). I guess we'd need to pre-allocate somehow (using `cpumask_var` and `struct page **`).

TBH I'd keep it for a later patch because it's not fully trivial, as we don't want to fail allocation during fault handling and there does not seem to be a fully exclusive code path (in case we would want to pre-alloc in some xpmem context).

Reply:
try to use percpu